ai/core/src/unitycatalog/ai/core/oss.py

```diff
@@ -0,0 +1,647 @@
+        value_python_type = column_type_to_python_type(
+            param_info.type_name, mapping=SQL_TYPE_TO_PYTHON_TYPE_MAPPING_UC_OSS
+        )
+        if not isinstance(value, value_python_type):
+            raise ValueError(
+                f"Parameter {param_info.name} should be of type {param_info.type_name} "
+                f"(corresponding python type {value_python_type}), but got {type(value)}"
+            )
+        validate_param(value, param_info.type_name, param_info.type_text)
+
+    @override
+    def _execute_uc_function(
+        self, function_info: FunctionInfo, parameters: Dict[str, Any], **kwargs: Any
+    ) -> FunctionExecutionResult:
+        if function_info.name in self.func_cache:
+            result = self.func_cache[function_info.name](**parameters)
+            return FunctionExecutionResult(format="SCALAR", value=str(result))
+        else:
+            python_function = dynamically_construct_python_function(function_info)
+            exec(python_function.function_def, self.func_cache)
+            try:
+                func = self.func_cache[function_info.name]
+
+                result = func(**parameters)
+
+                self.func_cache[function_info.name] = lru_cache()(func)
+
+                return FunctionExecutionResult(format="SCALAR", value=str(result))
+            except Exception as e:
+                return FunctionExecutionResult(error=str(e))
+
+    async def delete_function_async(
+        self,
+        function_name: str,
+        timeout: Optional[float] = None,
+    ) -> None:
+        """
+        Delete a function by its full name asynchronously.
+
+        Args:
+            function_name: The full name of the function to delete.
+                It should be in the format of "catalog.schema.function_name".
+            timeout: The timeout in seconds.
+
+        Returns:
+            None
+        """
+
+        await self.uc.delete_function(function_name, _request_timeout=timeout)
+
+    @override
+    @syncify_method
+    def delete_function(self, function_name: str, timeout: Optional[float] = None) -> None:
+        """
+        Delete a function by its full name.
+
+        This is a synchronous version of `delete_function_async`.
+
+        Args:
+            function_name: The full name of the function to delete.
+                It should be in the format of "catalog.schema.function_name".
+            timeout: The timeout in seconds.
+
+        Returns:
+            None
+        """
+
+        pass
+
+    @override
+    def to_dict(self) -> Dict[str, Any]:
+        return {k: getattr(self, k) for k in ["uc"] if getattr(self, k) is not None}
+
+
+class FunctionDefinition(NamedTuple):
+    """
+    A named tuple representing a dynamically constructed Python function.
+
+    Attributes:
+        function_def: The function definition as a string.
+        function_head: The function signature as a string.
```

Comment:
Seems this is not needed anymore :) We can drop this class

Reply:
Ahhhh thank you! Completely forgot to clean this up after removing the function_head logic :)